src/rdiff-backup-delete

```diff
@@ -225,23 +226,45 @@ def _remove_increments(path, dry_run):
 
 def _rmtree(path, dry_run):
     """
-    Custom implementation of shutil.rmtree() to support deletion of symlink.
+    Custom implementation of shutil.rmtree() to handle permission errors,
+    fifo and symlink deletion.
     """
+    if dry_run:
+        return
+
+    # Try to change the permissions of the file or directory to delete them.
+    def onerror(func, path, exc_info):
+        if exc_info[0] == PermissionError:
+            # Parent directory must allow rwx
+            if not os.access(os.path.dirname(path), os.W_OK | os.R_OK | os.X_OK):
+                os.chmod(os.path.dirname(path), 0o0700)
+            if not os.access(path, os.W_OK | os.R_OK):
+                os.chmod(path, 0o0600)
+            return func(path)
+        raise
+
     try:
         mode = os.lstat(path).st_mode
-    except os.error:
+    except FileNotFoundError:
```

Comment:
Generic remark: have you tested your code under Windows? I ask because, under Windows, I noticed recently, as I worked on the Windows specific issues raised on the alpha (e.g. long path one) that the exceptions thrown might be "surprising"... And `os.chmod` doesn't really work as expected under Windows...

Reply:
I have no way of testing this on Windows and I'm not sure the current implementation is compatible with Windows either. As part of this task, I just wanted to fix the permissions error on Linux. Making it work on Windows seams out of scope for this commit.

Reply:
Fair enough.